app/services/SalesforceService.scala

```diff
@@ -0,0 +1,99 @@
+package services
+
+import akka.agent.Agent
+import com.gu.membership.salesforce.Member.Keys
+import com.gu.membership.salesforce.{Authentication, MemberId, MemberRepository, Scalaforce}
+import com.typesafe.scalalogging.LazyLogging
+import configuration.Config
+import model.PersonalData
+import play.api.Logger
+import play.api.libs.concurrent.Akka
+import play.api.libs.json.{JsObject, Json}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.concurrent.{Await, Future}
+
+trait SalesforceService extends LazyLogging {
+
+  def createOrUpdateUser(personalData: PersonalData, userId: UserId): Future[Option[MemberId]]
+
+  def createSalesforceUserData(personalData: PersonalData): JsObject = {
+    Seq(Json.obj(
+      Keys.EMAIL -> personalData.email,
+      Keys.FIRST_NAME -> personalData.firstName,
+      Keys.LAST_NAME -> personalData.lastName,
+      Keys.MAILING_STREET -> personalData.address.street,
+      Keys.MAILING_CITY -> personalData.address.town,
+      Keys.MAILING_POSTCODE -> personalData.address.postcode,
+      Keys.MAILING_COUNTRY -> "United Kingdom",
+      Keys.ALLOW_MEMBERSHIP_MAIL -> true
+    )) ++ Map(
+      Keys.ALLOW_THIRD_PARTY_EMAIL -> Some(true),
+      Keys.ALLOW_GU_RELATED_MAIL -> Some(true)
+    ).collect { case (k, Some(v)) => Json.obj(k -> v) }
+  }.reduce(_ ++ _)
+
```

Comment:
Is there a specific reason why we do these intermediate transformations here (i.e. `.collect`, `.reduce`)? Why not just returning a Json object?

```scala
    Json.obj(
      Keys.EMAIL -> personalData.email,
      Keys.FIRST_NAME -> personalData.firstName,
      Keys.LAST_NAME -> personalData.lastName,
      Keys.MAILING_STREET -> personalData.address.street,
      Keys.MAILING_CITY -> personalData.address.town,
      Keys.MAILING_POSTCODE -> personalData.address.postcode,
      Keys.MAILING_COUNTRY -> "United Kingdom",
      Keys.ALLOW_MEMBERSHIP_MAIL -> true,
      Keys.ALLOW_THIRD_PARTY_EMAIL -> true,
      Keys.ALLOW_GU_RELATED_MAIL -> true)
```